lib/util/github/graphql/datasource-fetcher.ts

```diff
@@ -226,9 +227,14 @@ export class GithubGraphqlDatasourceFetcher<
     }
     const cacheNs = this.getCacheNs();
     const cacheKey = this.getCacheKey();
-    this._cacheStrategy = this.isPersistent
-      ? new GithubGraphqlPackageCacheStrategy<ResultItem>(cacheNs, cacheKey)
-      : new GithubGraphqlMemoryCacheStrategy<ResultItem>(cacheNs, cacheKey);
+    const cachePrivatePackages = GlobalConfig.get(
+      'cachePrivatePackages',
+      false,
+    );
+    this._cacheStrategy =
+      cachePrivatePackages || this.isPersistent
```

Comment:
How is isPersistence calculated? Could we end up caching stuff we shouldn't?

Reply:
It's determined by the `isRepoPrivate` flag returned together with the first page of results